hack/update-manifests.sh

```diff
@@ -1,22 +1,23 @@
 #!/bin/sh
 
-SRCROOT="$( cd "$(dirname "$0")/.." ; pwd -P )"
+set -e
+
+SRCROOT="$( CDPATH='' cd -- "$(dirname "$0")/.." && pwd -P )"
 AUTOGENMSG="# This is an auto-generated file. DO NOT EDIT"
 
 update_image () {
   if [ ! -z "${IMAGE_NAMESPACE}" ]; then
-    sed -i '' 's| image: \(.*\)/\(argocd.*\)| image: '"${IMAGE_NAMESPACE}"'/\2|g' ${1}
+    sed -i -- 's| image: \(.*\)/\(argocd.*\)| image: '"${IMAGE_NAMESPACE}"'/\2|g' "${1}"
```

Comment:
Not sure whether this is an BSD vs GNU thing, but the previous version failed for me on Linux. Hope this works on OSX, too?

Reply:
Just tried it -- it works on OS X, so you're good.

Reply:
Oops I spoke too soon, I think we need to follow something like:
https://stackoverflow.com/questions/5694228/sed-in-place-flag-that-works-both-on-mac-bsd-and-linux

to support in-place editing that works on both linux and mac